tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -34,17 +34,17 @@ class XlaGpuDeviceFactory : public DeviceFactory {
   Status CreateDevices(const SessionOptions& options, const string& name_prefix,
                        std::vector<std::unique_ptr<Device>>* devices) override;
   // Returns a set containing the device ids contained in visible_device_list or
-  // -1 if the string is empty.
-  static xla::StatusOr<std::set<int>> ParseVisibleDeviceList(
+  // nullopt if it is empty. It returns error in case of malformed configuration
+  // string.
+  static xla::StatusOr<absl::optional<std::set<int>>> ParseVisibleDeviceList(
       const string& visible_device_list);
 };
 
-xla::StatusOr<std::set<int>> XlaGpuDeviceFactory::ParseVisibleDeviceList(
-    const string& visible_device_list) {
+xla::StatusOr<absl::optional<std::set<int>>>
+XlaGpuDeviceFactory::ParseVisibleDeviceList(const string& visible_device_list) {
   std::set<int> gpu_ids;
-  if (visible_device_list.length() == 0) {
-    gpu_ids.insert(-1);
-    return gpu_ids;
+  if (visible_device_list.empty()) {
+    return absl::optional<std::set<int>>(absl::nullopt);
```

Comment:
I think `return {nullopt}` or `return {{nullopt}}` may work.

Reply:
That was what I had initially but gcc couldn't be able to instantiate StatusOr from just nullopt.